abcpy/ProbabilisticModel.py

```diff
@@ -0,0 +1,865 @@
+                return False
+            index = 0
+            i = 0
+            for s in range(2):
+                index=0
+                i=0
+                while (i < len(parameters[s])):
+                    while (not (isinstance(self.parents[s][index], ProbabilisticModel)) or self.parents[s][index].updated):
+                        index += 1
+                    #NOTE THIS IS NOT POSSIBLE, BECAUSE AN EMPTY LIST IS GIVEN FOR FIXED WHOLE LISTS!
+                    '''
+                    if (not(parameters[i])):
+                        parameter_value = self.parents[s][index].sample_from_distribution(1, rng=rng)
+                        if(isinstance(parameter_value[0], (list, np.ndarray))):
+                            parameter_value=parameter_value[0]
+                        for j in range(len(parameter_value)):
+                            self.parameter_values[s][index + j] = parameter_value[j]
+                        index += len(parameter_value)
+                        i += len(parameter_value)'''
+                    for j in range(self.parents[s][index].dimension):
+                        self.parameter_values[s][index + j] = parameters[s][i]
+                        i += 1
+                    index += self.parents[s][index].dimension
+            self.updated = True
+            return True
+
+
+    def pdf(self, x):
+        lower_bound = self.parents[0]
+        upper_bound = self.parents[1]
+        if (np.product(np.greater_equal(x, np.array(lower_bound)) * np.less_equal(x, np.array(upper_bound)))):
+            pdf_value = 1. / np.product(np.array(upper_bound) - np.array(lower_bound))
+        else:
+            pdf_value = 0.
+        return pdf_value
+
+class StochLorenz95(ProbabilisticModel):
+    """Generates time dependent 'slow' weather variables following forecast model of Wilks [1],
+        a stochastic reparametrization of original Lorenz model Lorenz [2].
+
+        [1] Wilks, D. S. (2005). Effects of stochastic parametrizations in the lorenz ’96 system.
+        Quarterly Journal of the Royal Meteorological Society, 131(606), 389–407.
+
+        [2] Lorenz, E. (1995). Predictability: a problem partly solved. In Proceedings of the
+        Seminar on Predictability, volume 1, pages 1–18. European Center on Medium Range
+        Weather Forecasting, Europe
+
+        Parameters
+        ----------
+        parameters: list
+            Contains the probabilistic models and hyperparameters from which the model derives.
+        initial_state: numpy.ndarray, optional
+            Initial state value of the time-series, The default value is None, which assumes a previously computed
+            value from a full Lorenz model as the Initial value.
+        n_timestep: int, optional
+            Number of timesteps between [0,4], where 4 corresponds to 20 days. The default value is 160.
+        """
+    def __init__(self, parameters, initial_state= None, n_timestep=160):
```

Comment:
The docstring (copied from the original master branch) says "number of time steps between [0,4], where 4 corresponds to 20 days). The constructor has a default value for n_timestep of 160. This doesn't really make sense to me.

Reply:
I think this you have to ask Rito -- he knows the model.